src/theory/quantifiers/inst_strategy_enumerative.cpp

```diff
@@ -175,189 +176,52 @@ void InstStrategyEnum::check(Theory::Effort e, QEffort quant_e)
   }
 }
 
-bool InstStrategyEnum::process(Node f, bool fullEffort, bool isRd)
+bool InstStrategyEnum::process(Node quantifier, bool fullEffort, bool isRd)
 {
-  // ignore if constant true (rare case of non-standard quantifier whose body is
-  // rewritten to true)
-  if (f[1].isConst() && f[1].getConst<bool>())
+  // ignore if constant true (rare case of non-standard quantifier whose body
+  // is rewritten to true)
+  if (quantifier[1].isConst() && quantifier[1].getConst<bool>())
   {
     return false;
   }
-  unsigned final_max_i = 0;
-  std::vector<unsigned> maxs;
-  std::vector<bool> max_zero;
-  bool has_zero = false;
-  std::map<TypeNode, std::vector<Node> > term_db_list;
-  std::vector<TypeNode> ftypes;
-  TermDb* tdb = d_quantEngine->getTermDatabase();
-  QuantifiersState& qs = d_quantEngine->getState();
-  // iterate over substitutions for variables
-  for (unsigned i = 0; i < f[0].getNumChildren(); i++)
+
+  TermTupleEnumeratorContext ttec;
+  ttec.d_quantEngine = d_quantEngine;
+  ttec.d_rd = d_rd;
+  ttec.d_fullEffort = fullEffort;
+  ttec.d_increaseSum = options::fullSaturateSum();
+  ttec.d_isRd = isRd;
+  std::unique_ptr<TermTupleEnumeratorInterface> enumerator(
+      mkTermTupleEnumerator(quantifier, &ttec));
+  std::vector<Node> terms;
+  std::vector<bool> failMask;
+  Instantiate* const ie = d_quantEngine->getInstantiate();
```

Comment:
```suggestion
  Instantiate* ie = d_quantEngine->getInstantiate();
```
As a style preference, I don't add const to a pointer like this.

Reply:
I feel like this is somewhat inconsistent. Wouldn't it make more sense if either you never use `const` or you try to use it as much as possible? Note, however, that `const` is already used in the code base in this particular situation (see `grep -r '\* \<const\>' src `).

Reply:
You're right, there are some inconsistencies on this point (although most code that uses this style are old).  This is a minor point, but the reason why I don't like `const` in this case is that non-constant methods are being called on `ie` (e.g. https://github.com/CVC4/CVC4/pull/6030/files#diff-9d139abfe6d05092d511906633375cde3335cb3bb8f00be3d72f4b46a066e23aR210), so it's a bit confusing to say we have a "const pointer" here, I guess this means that the value of the pointer does not change (but the underlying class may)?

FYI, the reference on style guidelines is here: https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines although subtle points like this are generally up for debate.

Reply:
Yes, you're right it just means that the local variable isn't going to change while the contents pointed to may change. It's just as saying `const int i  = 0;`. Anyhow, I'll get rid of it.